tm2/pkg/std/memfile.go

```diff
@@ -37,14 +38,32 @@ func (mempkg *MemPackage) IsEmpty() bool {
 	return len(mempkg.Files) == 0
 }
 
-const rePathPart = `[a-z][a-z0-9_]*`
-
 var (
-	rePkgName      = regexp.MustCompile(`^[a-z][a-z0-9_]*$`)
-	rePkgOrRlmPath = regexp.MustCompile(`gno\.land/(?:p|r)(?:/` + rePathPart + `)+`)
-	reFileName     = regexp.MustCompile(`^([a-zA-Z0-9_]*\.[a-z0-9_\.]*|LICENSE|README)$`)
+	rePathPart = regexp.MustCompile(`^[a-z0-9_]+$`)
+	rePkgName  = regexp.MustCompile(`^[a-z][a-z0-9_]*$`)
+	reFileName = regexp.MustCompile(`^([a-zA-Z0-9_]*\.[a-z0-9_\.]*|LICENSE|README)$`)
 )
 
+func validatePkgOrRlmPath(path string) error {
+	parts := strings.Split(path, "/")
+	if len(parts) < 3 {
+		return errors.New("path must be in the format [gno.land/{p|r}/path/...]")
+	}
+	if parts[0] != "gno.land" {
+		return errors.New("invalid domain, must be gno.land")
+	}
+	if parts[1] != "p" && parts[1] != "r" {
+		return fmt.Errorf("must be 'p' or 'r'")
+	}
```

Comment:
I don't think this is an improvement over the past idea that we can use a single regex to validate the "first 2 parts" of the path, `rePkgOrRlmPath`.

You can do this with simply 2 regex matches, one for matching the `gno.land` part and `p` or `r`, and one for matching everything else (after the second `/`).
You can even combine it into a single regex, if we're fine with that kind of readability.

This way, there is no ambiguity over whether there are exactly 3 sections to the realm / package path, and what their format is

Reply:
I still believe having non-regex solution provides better readability and control over the checks. 

Modified the solution to use regex only. Thanks.